migrator-tool/src/main/scala/za/co/absa/spline/migrator/DataLineageToExecPlanWithEventConverter.scala

```diff
@@ -89,29 +89,43 @@ class DataLineageToExecPlanWithEventConverter(lineage: DataLineage) {
     (opWrite, opReads, opOthers)
   }
 
-  private def convertWriteOperation(opWrite: op.Write): WriteOperation =
+  private def convertWriteOperation(lineage: DataLineage, opWrite: op.Write): WriteOperation =
     WriteOperation(
       outputSource = opWrite.path,
       append = opWrite.append,
       id = operationIdByDatasetUUID(opWrite.mainProps.output),
       childIds = opWrite.mainProps.inputs.map(operationIdByDatasetUUID),
-      schema = None, // todo: to be implemented in https://github.com/AbsaOSS/spline/issues/258
+      schema = findOutputSchema(lineage, opWrite),
       params = Map(
         OperationParams.Name -> opWrite.mainProps.name,
         OperationParams.DestinationType -> opWrite.destinationType
       ))
 
-  private def convertReadOperation(opRead: op.Read): ReadOperation =
+  private def convertReadOperation(lineage: DataLineage, opRead: op.Read): ReadOperation =
     ReadOperation(
       inputSources = opRead.sources.map(_.path),
       id = operationIdByDatasetUUID(opRead.mainProps.output),
-      schema = None, // todo: to be implemented in https://github.com/AbsaOSS/spline/issues/258
+      schema = findOutputSchema(lineage, opRead),
       params = Map(
         OperationParams.Name -> opRead.mainProps.name,
         OperationParams.SourceType -> opRead.sourceType
       ))
 
-  private def convertOtherOperation(opOther: op.Operation): DataOperation = {
+  private def findOutputSchema(dataLineage: DataLineage, operation: op.Operation): Map[String, Seq[Map[String, String]]] = {
+    val metaDataset: MetaDataset = dataLineage.datasets.find((dts: MetaDataset) => dts.id == operation.mainProps.output)
+      .getOrElse(throw new IllegalArgumentException(
+        s"Operation output id ${operation.mainProps.output} not found in datasets of dataLineage ${dataLineage.id}"))
+    val attributes = metaDataset.schema.attrs.map(attrId => {
```

Comment:
what about algorithmic complexity of this piece?
I'd suggest creating a map of `AttrId -> Attribute` first and then apply it to a `schema.attrs`. E.g.
```suggestion
val attrById = dataLineage.attributes.map(a => a.id -> a).toMap
    val schemaAttributes =
      metaDataset.schema.attrs
        .map(attrById)
        .map(attr => Map(
          "name" -> attr.name,
          "dataTypeId" -> attr.dataTypeId.toString))
```

Reply:
Also, using handling options seems unnecessary here. We assume the integrity and correctness of the lineages we process.